flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/combine/SortMergeReader.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.storage.filestore.combine;
+
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.storage.filestore.KeyValue;
+import org.apache.flink.table.storage.filestore.utils.RecordIterator;
+import org.apache.flink.table.storage.filestore.utils.RecordReader;
+
+import javax.annotation.Nullable;
+
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.PriorityQueue;
+
+/**
+ * This reader is to read a list of {@link RecordReader}, which is already sorted by key and
+ * sequence number, and perform a sort merge algorithm. Note that values are not combined at this
+ * phase.
+ */
+public class SortMergeReader implements RecordReader<KeyValue> {
+    private final List<RecordReader<KeyValue>> sortedReaders;
+    private final Comparator<RowData> userKeyComparator;
+    private PriorityQueue<Element> minHeap;
+    private Element current;
+
+    public SortMergeReader(
+            List<RecordReader<KeyValue>> sortedReaders, Comparator<RowData> userKeyComparator) {
+        this.sortedReaders = sortedReaders;
+        this.userKeyComparator = userKeyComparator;
+    }
+
+    @Nullable
+    @Override
+    public RecordIterator<KeyValue> readBatch() throws IOException {
+        if (sortedReaders.isEmpty()) {
+            return null;
+        }
+        if (Objects.isNull(minHeap)) {
+            heapify();
```

Comment:
initMinHeap?

Reply:
+1. 
Take `PriorityQueue#initElementsFromCollection` for reference, change `heapify` to `initElements` and `supply` to `supplyElements` maybe better.